src/Server/AwsCloudFormationTemplates.fs

```diff
@@ -12,481 +9,221 @@ open Shared
+        result {
+            let! _ = requirePart "IpProtocol" data
+            let! _ = requirePart "GroupId" data
+            let! importId = 
+                match tryFindSecurityGroupRule data context.securityGroupEgressRules with
+                | Some importId -> Ok importId
+                | None -> Error $"No matching security group rule found for {resource.resourceType} named {resource.logicalId}"
+            return importId
+        }
 
-    "AWS::ECS::Service" => {
-        pulumiType = getPulumiType "AWS::ECS::Service"
-        importIdentityParts = ["Id"]
-        delimiter = "/"
-        remapFunc = remapFromIdAsArn
-        validatorFunc = validateFromImportIdentityParts
-    }
+    "AWS::ECS::Service" => fun resource data context ->
+        result {
+            let! physicalId = requirePart "Id" data
+            let! importId = 
+                match physicalId.Split("/") with
+                | parts when parts.Length > 1 -> 
+                    parts
+                    |> Array.skip 1
+                    |> String.concat "/"
+                    |> Ok
+                | _ -> 
+                    Error $"Invalid Id: {physicalId}"
+
+            return importId
+        }
 
-    "AWS::ElasticLoadBalancingV2::ListenerCertificate" => {
-        pulumiType = getPulumiType "AWS::ElasticLoadBalancingV2::ListenerCertificate"
+    "AWS::ElasticLoadBalancingV2::ListenerCertificate" => defaultImportIdentity {
         importIdentityParts = ["ListenerArn"; "Certificates"]
         delimiter = "_"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
 
-    "AWS::Lambda::Permission" => {
-        pulumiType = getPulumiType "AWS::Lambda::Permission"
+    "AWS::Lambda::Permission" => defaultImportIdentity {
         importIdentityParts = ["FunctionName"; "Id"]
         delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
 
-    "AWS::Lambda::LayerVersionPermission" => {
-        pulumiType = getPulumiType "AWS::Lambda::LayerVersionPermission"
-        importIdentityParts = ["LayerVersionArn"]
-        delimiter = ","
-        remapFunc = remapLayerVersionPermission
-        validatorFunc = validateFromImportIdentityParts
-    }
+    "AWS::Lambda::LayerVersionPermission" => fun resource data context ->
+        result {
+            let! layerVersionArn = requirePart "LayerVersionArn" data
+            let! importId = 
+                // rewrite "part1:part2:partN:version" 
+                // into "part1:part2:partN,version"
+                match layerVersionArn.Split(":") with
+                | layerVersionArnParts when layerVersionArnParts.Length > 1 -> 
+                    let layerVersion = layerVersionArnParts[layerVersionArnParts.Length - 1]
+                    let layerArn = 
+                        layerVersionArnParts
+                        |> Array.except [| layerVersion |]
```

Comment:
I also struggled to find a pipelineable function that would return all but the last element - I guess this works but Id rather target that element more specifically

Reply:
Fixed ✅